backend/audit/test_schemas.py

```diff
@@ -1,20 +1,264 @@
 # Even though the schemas are not Django views or modules etc., we test them
 # here for CI/CD integration.
 import json
+import string
+
 from pathlib import Path
 from django.test import SimpleTestCase
-from jsonschema import exceptions, validate
+from jsonschema import exceptions, validate as jsonschema_validate, FormatChecker
+from random import choice, randrange
 
 # Simplest way to create a new copy of simple case rather than getting
 # references to things used by other tests:
 jsoncopy = lambda v: json.loads(json.dumps(v))
 
+
+# wrap the validate function to include a format checker
+def validate(instance, schema):
+    return jsonschema_validate(instance, schema, format_checker=FormatChecker())
```

Comment:
the `format_checker` bit here is necessary if you want `jsonschema` to validate type-formatted strings (e.g. `email`, `date`, etc). We may want to wrap the `validate` call in a utility function so that we don't have to remember to do this in each case